hugr/src/hugr/rewrite/outline_cfg.rs

```diff
@@ -248,106 +249,159 @@ pub enum OutlineCfgError {
 mod test {
     use std::collections::HashSet;
 
-    use crate::algorithm::nest_cfgs::test::{
-        build_cond_then_loop_cfg, build_conditional_in_loop, build_conditional_in_loop_cfg, depth,
-    };
     use crate::builder::{
-        Container, Dataflow, DataflowSubContainer, HugrBuilder, ModuleBuilder, SubContainer,
+        BlockBuilder, BuildError, CFGBuilder, Container, Dataflow, DataflowSubContainer,
+        HugrBuilder, ModuleBuilder,
     };
     use crate::extension::prelude::USIZE_T;
-    use crate::extension::PRELUDE_REGISTRY;
+    use crate::extension::{ExtensionSet, PRELUDE_REGISTRY};
     use crate::hugr::views::sibling::SiblingMut;
     use crate::hugr::HugrMut;
-    use crate::ops::handle::{BasicBlockID, CfgID, NodeHandle};
+    use crate::ops::constant::Value;
+    use crate::ops::handle::{BasicBlockID, CfgID, ConstID, NodeHandle};
     use crate::types::FunctionType;
-    use crate::{type_row, HugrView, Node};
+    use crate::{type_row, Hugr, HugrView, Node};
     use cool_asserts::assert_matches;
     use itertools::Itertools;
 
     use super::{OutlineCfg, OutlineCfgError};
 
+    //      /-> left --\
+    // entry            > merge -> head -> tail -> exit
+    //      \-> right -/             \-<--<-/
+    // Result is Hugr plus merge and tail blocks
+    fn build_cond_then_loop_cfg() -> Result<(Hugr, BasicBlockID, BasicBlockID), BuildError> {
```

Comment:
Lots of test lines below try to recover `entry, left, right, head, exit` from the `merge, tail` returned by this function. You could consider building a struct instead which I think would make the tests below less noisy:
```rust
//      /-> left --\
// entry            > merge -> head -> tail -> exit
//      \-> right -/             \-<--<-/
struct CondThenLoopCFG {
   h: Hugr,
   entry: BasicBlockId,
   left: BasicBlockId,
   right: BasicBlockId,
   ...
}

impl CondThenLoopCFG {
   fn new() -> Result<Self, BuildError> {
      ...
   }
}

```

Reply:
Yes, great idea, a side benefit of using the same example for many tests

Reply:
Done with an rstest fixture, the extra benefit of which is slight, but perhaps non-zero